src/lib.rs

```diff
@@ -621,6 +621,15 @@ impl<T> From<(T, Vec<T>)> for NonEmpty<T> {
     }
 }
 
+impl<T> IntoIterator for NonEmpty<T> {
+    type Item = T;
+    type IntoIter = std::iter::Chain<std::iter::Once<T>, std::vec::IntoIter<Self::Item>>;
```

Comment:
Can we import some of this stuff to make this line more readable? Maybe we import at least `std::{iter, vec}` at the top level?